ding/torch_utils/optimizer_helper.py

```diff
@@ -5,6 +5,33 @@
 from typing import Union, Iterable, Tuple, Callable
 
 
+def calculate_grad_norm(parameters, norm_type=2):
+    parameters = list(filter(lambda p: p.grad is not None, parameters))
+    if parameters == []:
+        parameters = 0
```

Comment:
why do this assignment

Reply:
For the first time the forward learn code is executed,  the grad is none. To avoid error, we assign the parameters to 0.